client/v3/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2024 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"context"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// New creates KubernetesClient from config.
+// Caller is responsible to call Close() to clean up client.
+func New(cfg clientv3.Config) (*KubernetesClient, error) {
+	c, err := clientv3.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return fromClient(c, c.Close), nil
+}
+
+// FromClient creates KubernetesClient from existing client.
+// Caller can still call Close() however, it's not needed as long caller should call Close() on the original client.
+func FromClient(c *clientv3.Client) *KubernetesClient {
+	return fromClient(c, nil)
```

Comment:
Or you can remove the `fromClient`, and move the implementation into `FromClient`
```suggestion
	return fromClient(c)
```

Reply:
Done